connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -742,33 +750,49 @@ protected static SchemaAndValue parse(Parser parser, boolean embedded) throws No
                 return new SchemaAndValue(Schema.STRING_SCHEMA, sb.toString());
             }
         }
-        if (parser.canConsume(TRUE_LITERAL)) {
+
+        if (canParseSingleTokenLiteral(parser, embedded, NULL_VALUE)) {
+            return null;
+        }
+        if (canParseSingleTokenLiteral(parser, embedded, TRUE_LITERAL)) {
             return TRUE_SCHEMA_AND_VALUE;
         }
-        if (parser.canConsume(FALSE_LITERAL)) {
+        if (canParseSingleTokenLiteral(parser, embedded, FALSE_LITERAL)) {
             return FALSE_SCHEMA_AND_VALUE;
         }
+
         int startPosition = parser.mark();
+
         try {
             if (parser.canConsume(ARRAY_BEGIN_DELIMITER)) {
+                if (parser.canConsume(ARRAY_END_DELIMITER)) {
+                    Schema emptyListSchema = SchemaBuilder.arrayOfNull().build();
+                    return new SchemaAndValue(emptyListSchema, new ArrayList<>());
```

Comment:
I opted to stick with allocating a new object each time since that's the current behavior of this class, and changing to an immutable return value may end up breaking existing applications that rely on a mutable return value.
FWIW, if I'd written this from the ground up I'd probably be inclined to go the way of `Collections::emptyList`, but it seems too late now to make that change.